### PR TITLE
Feature/switch attributes screen

### PR DESCRIPTION
**Pull request by Vin-RM**
Add Screen to Switch or Add an Attribute for Read and Propose Attribute Requests.

### CODE REVIEW

File: packages/renderers/lib/src/request/request_renderer.dart [hunk 1]
```diff
@@ -47,7 +51,9 @@ class RequestRenderer extends StatelessWidget {
         );
       }).toList();
 
-      return ListView(children: responseItems);
+      return SingleChildScrollView(
+        child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: responseItems),
+      );
```
Comment (jkoenig134) [inline packages/renderers/lib/src/request/request_renderer.dart:56]: please change back.

File: packages/renderers/lib/src/request/request_renderer.dart [hunk 2]
```diff
@@ -68,12 +74,14 @@ class RequestRenderer extends StatelessWidget {
...
+        openAttributeScreen: openAttributeScreen,
         requestStatus: request.status,
         currentAddress: currentAddress,
       );
     }).toList();
 
-    return ListView(children: requestItems);
+    return SingleChildScrollView(
+      child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: requestItems),
+    );
```
Comment (jkoenig134) [inline packages/renderers/lib/src/request/request_renderer.dart:85]: see above.

File: packages/renderers/lib/src/request/request_renderer.dart [hunk 3]
```diff
@@ -11,15 +11,19 @@ class RequestRenderer extends StatelessWidget {
   final RequestRendererController? controller;
   final LocalRequestDVO request;
   final RequestValidationResultDTO? validationResult;
-  final Future<AbstractAttribute> Function({required String valueType})? selectAttribute;
   final String currentAddress;
+  final Future<AbstractAttribute?> Function({
+    required String valueType,
+    required List<AbstractAttribute> attributes,
+    ValueHints? valueHints,
+  })? openAttributeScreen;
```
Comment (jkoenig134) [inline packages/renderers/lib/src/request/request_renderer.dart:19]: you can extract this to a typedef (in an own file)

File: packages/renderers/lib/src/request/request_item_renderer/widgets/open_attribute_screen.dart [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import 'package:enmeshed_types/enmeshed_types.dart';
+
+typedef OpenAttributeScreen = Future<AbstractAttribute?> Function({
```
Comment (jkoenig134) [inline packages/renderers/lib/src/request/request_item_renderer/widgets/open_attribute_screen.dart:3]:
```suggestion
typedef OpenAttributeScreenFunction = Future<AbstractAttribute?> Function({
```

File: packages/renderers/lib/src/request/request_item_renderer/decidable/decidable_read_attribute_request_item_renderer.dart [hunk 5]
```diff
@@ -154,32 +149,62 @@ class _DecidableReadAttributeRequestItemRendererState extends State<DecidableRea
...
+      final ProcessedIdentityAttributeQueryDVO query => query.results,
+      final ProcessedRelationshipAttributeQueryDVO query => query.results,
+      final ProcessedThirdPartyRelationshipAttributeQueryDVO query => query.results,
+      final ProcessedIQLQueryDVO query => query.results,
+    };
+
+    return results.map((result) => (id: result.id, attribute: result.content)).toList();
   }
 
+  // TODO(Vin-RM): if there is an ENABLE checkbox why don't we need a DISABLE checkbox?
```
Comment (jkoenig134) [inline packages/renderers/lib/src/request/request_item_renderer/decidable/decidable_read_attribute_request_item_renderer.dart:207]: @Vin-RM could you please look into this? It looks sus that we only enable the checkbox, but not disable when the input gets invalid again.

**Review by jkoenig134 — DISMISSED** [review]

**Review by jkoenig134 — APPROVED** [review]

File: packages/renderers/example/lib/pages/account_requests/widgets/attribute_screen.dart [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+      appBar: AppBar(
+        title: TranslatedText('i18n://dvo.attribute.name.${widget.attributeTitle}'),
+      ),
+      body: Column(
+        children: [
+          Padding(
+            padding: const EdgeInsets.all(16),
+            child: Column(
+              children: [
+                const Text('Share an entry you have already created with your contact or create a new one for it.'),
```
Comment (Siolto) [inline packages/renderers/example/lib/pages/account_requests/widgets/attribute_screen.dart:46]: Translation missing

File: packages/renderers/example/lib/pages/account_requests/widgets/attribute_screen.dart [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                  children: [
+                    const Text('My Entries', style: TextStyle(fontWeight: FontWeight.bold)),
```
Comment (Siolto) [inline packages/renderers/example/lib/pages/account_requests/widgets/attribute_screen.dart:50]: Translation

File: packages/renderers/example/lib/pages/account_requests/widgets/attribute_screen.dart [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+                    TextButton.icon(
+                      onPressed: () {},
+                      icon: const Icon(Icons.add, size: 16),
+                      label: const Text('Add Entry'),
```
Comment (Siolto) [inline packages/renderers/example/lib/pages/account_requests/widgets/attribute_screen.dart:54]: translation

File: packages/renderers/example/lib/pages/account_requests/widgets/attribute_screen.dart [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+              ],
+            ),
+            child: Padding(
+              padding: const EdgeInsets.all(16),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.end,
+                children: [
+                  TextButton(
+                    onPressed: () => Navigator.pop(context),
+                    child: const Text('Cancel'),
```
Comment (Siolto) [inline packages/renderers/example/lib/pages/account_requests/widgets/attribute_screen.dart:123]: translation

File: packages/renderers/example/lib/pages/account_requests/widgets/attribute_screen.dart [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+                  ),
+                  FilledButton(
+                    style: OutlinedButton.styleFrom(minimumSize: const Size(100, 36)),
+                    onPressed: () => Navigator.pop(context, selectedOption),
+                    child: const Text('Select'),
```
Comment (Siolto) [inline packages/renderers/example/lib/pages/account_requests/widgets/attribute_screen.dart:128]: translation

File: packages/renderers/example/lib/pages/account_requests/widgets/attribute_screen.dart (hunk 6 above)
Comment (jkoenig134) [reply packages/renderers/example/lib/pages/account_requests/widgets/attribute_screen.dart:46]: this is just the example app, we don't translate there. Same for the other comments.

**Pull request merged**